src/custom/pages/Profile/index.tsx

```diff
@@ -63,7 +75,23 @@ export default function Profile() {
 
   const hasCowBalance = cow && !cow.equalTo(0)
   const hasVCowBalance = total && !total.equalTo(0)
-  const hasVested = vested && !vested.equalTo(0)
+  const hasVestedBalance = vested && !vested.equalTo(0)
+
+  const isSwapDisabled = Boolean(!hasVestedBalance || swapVCowStatus !== SwapVCowStatus.INITIAL)
+
+  const { swapCallback } = useSwapVCowCallback()
+
+  const handleVCowSwap = () => {
+    setSwapVCowStatus(SwapVCowStatus.ATTEMPTING)
+    swapCallback()
+      .then(() => {
+        setSwapVCowStatus(SwapVCowStatus.SUBMITTED)
+      })
+      .catch((error) => {
+        setSwapVCowStatus(SwapVCowStatus.INITIAL)
+        console.error('[Profile::index::swapVCowCallback]::error', error)
+      })
```

Comment:
I don't think we should model this as a global state.

What happens if an user swaps and in next block, before the swapping is done, they want to swap again?